JobService panics on startup when Core is not yet ready

JobService panics on startup at `main.go:50` when Core's HTTP server isn't ready yet:

```
panic: failed to load configuration, error: Get "http://core:8080/api/v2.0/internalconfig": dial tcp ...:8080: connect: connection refused
```

This happens because `cfgLib.DefaultMgr().Load()` (which fetches config from Core via REST) is called without any retry logic. Docker Compose's `depends_on` only waits for the container to start, not for the HTTP server inside it to be ready.

## Root Cause

In `src/jobservice/main.go`, the config load on line 49 fails immediately if Core hasn't started its HTTP listener yet:

```go
if err := cfgLib.DefaultMgr().Load(context.Background()); err != nil {
    panic(fmt.Sprintf("failed to load configuration, error: %v", err))
}
```

## Proposed Fix

Wrap the config load in `retry.Retry()` with exponential backoff (500ms initial, 10s max, 5min timeout). This mirrors the exact pattern Core already uses to wait for JobService health at startup (`src/core/main.go` ~line 289).

## Impact

- **Without fix**: JobService crashes on startup in any environment where Core takes more than a few milliseconds to become ready (common with Docker Compose, Kubernetes)
- **With fix**: JobService gracefully waits up to 5 minutes for Core, logging retry attempts so operators can see what's happening

--- a/src/jobservice/main.go
+++ b/src/jobservice/main.go
@@ -19,6 +19,7 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"time"
 
 	"github.com/goharbor/harbor/src/common"
 	"github.com/goharbor/harbor/src/jobservice/common/utils"
@@ -29,6 +30,8 @@ import (
 	"github.com/goharbor/harbor/src/jobservice/runtime"
 	"github.com/goharbor/harbor/src/lib"
 	cfgLib "github.com/goharbor/harbor/src/lib/config"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/retry"
 	tracelib "github.com/goharbor/harbor/src/lib/trace"
 	_ "github.com/goharbor/harbor/src/pkg/accessory/model/base"
 	_ "github.com/goharbor/harbor/src/pkg/accessory/model/cosign"
@@ -46,7 +49,16 @@ func main() {
 	lib.StartPprof()
 
 	cfgLib.DefaultCfgManager = common.RestCfgManager
-	if err := cfgLib.DefaultMgr().Load(context.Background()); err != nil {
+	if err := retry.Retry(func() error {
+		return cfgLib.DefaultMgr().Load(context.Background())
+	},
+		retry.InitialInterval(500*time.Millisecond),
+		retry.MaxInterval(10*time.Second),
+		retry.Timeout(5*time.Minute),
+		retry.Callback(func(err error, sleep time.Duration) {
+			log.Debugf("failed to load configuration from core, retry after %s: %v", sleep, err)
+		}),
+	); err != nil {
 		panic(fmt.Sprintf("failed to load configuration, error: %v", err))
 	}
 

